### PR TITLE
Introducing separate branch to conduct code review

### PR DESCRIPTION
**Pull request by peterdutey**
Hi @AnikaC-git 
Here you go, I look forward to your comments! 😃
(this is the correct PR - previous one was on wrong branch)
Thank you
Peter

### CODE REVIEW

File: DESCRIPTION
Comment (AnikaC-git) [inline DESCRIPTION:12]: Sorry this is tiny really, but not having seen the code but I wonder whether it is really "manipulate SNOMED-CT", which I understand to mean change the ontology as such

File: DESCRIPTION
Comment (AnikaC-git) [inline DESCRIPTION:3]: Small observation, in other places (e.g. rest-api.R) SNOMED-CT is written without hyphen

**Review by AnikaC-git — APPROVED** [review]
I would like to start by saying that it's wonderful to see this R wrapper progressing and I can see how it's beneficial to those that have experience with R -- perhaps on an analytical side -- but not so much with talking to SNOMED endpoints. I hope that there will be a great uptake of it in no time.

Overall the code looks fine to me and the separation into individual modules. I particularly liked that you have test cases added for the individual functions and tried to be as comprehensive as possible with your tests. The only small point that I mention is that it might perhaps be good to consider what the purpose of a test is as it could be a vehicle to help you spot "misbehaviour" faster instead of just adding additional secure that everything is working well. I note that for a project this size, it's probably not necessary but could be relevant for bigger projects. 

What I haven't done while going through is checked whether all requirements for a submission to rOpenSci (as available through their documentation) are fulfilled. If this is something you'd like me to do, please let me know. 

One question I had while looking through is and I might have missed it, is whether you'd want to release the code under a specific license. While I write this I notice that I am not sure myself whether we have a standard license at UCL, but I will ask in the wider team to clarify.

While I looked through the files, it seemed that the structure of the files had changed somewhere and I am wondering whether I looked at the most recent version. Now being more familiar with it, a second review should be much faster. 

Any questions in relation to my comments, please let me know.

**Comment by AnikaC-git** [comment]
Please ignore my comment about the license, I just saw it. I wonder whether it might make sense to mention it in the documentation somewhere?

File: R/zzz.R
Comment (peterdutey) [reply R/zzz.R:3]: Good point - revamped documentation [here](https://festive-agnesi-07b187.netlify.app/reference/snomedizer_options.html), hopefully a lot clearer now.

File: R/zzz.R
Comment (peterdutey) [reply R/zzz.R:22]: Good point - revamped documentation [here](https://festive-agnesi-07b187.netlify.app/reference/snomedizer_options.html), hopefully a lot clearer now.

File: R/utils.R
Comment (peterdutey) [reply R/utils.R:72]: Good point - revamped documentation [here](https://festive-agnesi-07b187.netlify.app/reference/snomedizer_options.html), hopefully a lot clearer now.

File: R/utils.R
Comment (peterdutey) [reply R/utils.R:79]: I may be missing something but don't think so - this only triggers is all three inputs are NULL. Happy to improve if that's an awkward behaviour.

File: R/utils.R
Comment (peterdutey) [reply R/utils.R:398]: Fixed in https://github.com/ramses-antibiotics/snomedizer/pull/15/commits/99cfd9e305daf68fa32688e1e8a96e270d372b93#

File: R/utils.R
Comment (peterdutey) [reply R/utils.R]: fixed https://github.com/ramses-antibiotics/snomedizer/blob/e804e02c18b949aa414e02ed6c75507009e7ab04/R/utils.R#L383

File: R/utils.R
Comment (peterdutey) [reply R/utils.R]: Agree - fixed https://github.com/ramses-antibiotics/snomedizer/blob/e804e02c18b949aa414e02ed6c75507009e7ab04/R/utils.R#L362-L364

File: R/utils.R
Comment (peterdutey) [reply R/utils.R:142]: agreed - fixed in https://github.com/ramses-antibiotics/snomedizer/commit/610b5db07ecde56265c0685187fa3b3ffe56986a

File: man/concepts_descendants.Rd
Comment (peterdutey) [reply man/concepts_descendants.Rd:42]: this function will return a nested list, with one entry for each parent conceptId provided in input.
So in case of overlap the descendants will appear as many times as there are parents.

File: tests/testthat/test.rest-api.R
Comment (peterdutey) [reply tests/testthat/test.rest-api.R:1]: Will do some more work on tests and respond in due time.

File: tests/testthat/test.wrapper.R
Comment (peterdutey) [reply tests/testthat/test.wrapper.R:2]: Will do some more work on tests and respond in due time.

**Comment by peterdutey** [comment]
Hi @AnikaC-git 

A massive thank you for the code review. This is really helpful and much appreciated 👍 👍 👍 

A few notes below

1. I've made changes to `develop` and referenced them in response to your questions. The only thing I would welcome re-review is the documentation for the option setter, which is packed in ?snomedizer-options. Perhaps easiest to look at the HTML rendering [at this link](https://festive-agnesi-07b187.netlify.app/reference/snomedizer_options.html)

2. In reference to your comment cited below, sorry you're having issues. I'm not certain what is going on, but it is clear I screwed up the branches pretty horridly on this occasion. I am fixing it little by little. Perhaps check you are on the `develop` branch? `utils.r` is definitely here to stay. Hopefully it will no longer be an issue once I've merged the final vignette into `develop`.

> 
> Other than that, has the file structure changed? As in it seems that the utils.R is gone and the code is now in rest-api.R? The problem this restructuring has is that it does not automically seem to update the PR and I seem to be unable to comment on the files as they are now, rather how they were when the PR was submitted.

3. I have now merged the main vignette into develop. It is meant to give a high-speed tour of what one can do with snomed. The success criteria is that the reader should immediately grasp the potential use case of each of the 4 examples before beginning to understand how to perform the task, so they are eager to learn more. I tried to come up with four interesting examples, but you may have better ones. You can see the HTML rendering [at this link](https://festive-agnesi-07b187.netlify.app/articles/snomedizer.html). I think it could still be made more accessible (not sure the point/use case of every example is made obvious from sentence 1).

4. Thanks to your comments I now plan to: (a) strengthen the tests as suggested; (b) work on the remaining two vignettes (one on taxonomies, one demonstrating rough-and-ready entity recognition), (c) get you a first draft of the paper, and (d) run through the rOpenSci checklist. Then I'll apply for rOpenSci review. I'll be in touch about (a)-(d) in due time.

Thank you :)

Peter

**Pull request closed without merging**
